mineinabyss-components/src/main/kotlin/com/mineinabyss/components/guilds/GuildConfig.kt

```diff
@@ -0,0 +1,14 @@
+package com.mineinabyss.components.guilds
+
+import kotlinx.serialization.Serializable
+
+
+/**
+ * @property maxLength The maximum length of a guild name.
+ * @property bannedWords List for blocking certain words from guild names.
+ */
+@Serializable
+class GuildConfig (
```

Comment:
Wouldnt that not work unless core depends on feature?
Which if i do causes a circular dependency?

![image](https://user-images.githubusercontent.com/62521371/149522341-2ceb1c6e-00e3-4770-a69f-d3b5c76d6550.png)

Reply:
This shouldn't be in core anyways.

Reply:
Put it in core since thats where the database stuff was. So having extension functions seemed natural.
So the guildfilter is in the createguild func
Can move it if u want tho but

Reply:
Yep keep any feature specific access stuff within that feature too. In the future we might do modules where different modules need to explicitly depend on others so you want everything related close together.

Reply:
This is still an issue, please move the values here into `GuildFeature` and pass a reference to it into any listeners it creates (i.e. `class SomeGuildListener(val feature: GuildFeature)` then `feature.maxLength` to get that.)

If some things become impossible to do with that, let me know and ill figure something out.

Reply:
The only place this currently fails then is when referencing in MIAConfig.
![image](https://user-images.githubusercontent.com/62521371/150509559-3488904a-c6b6-4263-8f12-5b6ee9752985.png)

Needs a dependency on features then to be able to get those values iguess tho it causes circular dep?

Reply:
@0ffz comment?